### PR TITLE
Added guest required twig tag.

### PR DESCRIPTION
**Pull request by wrux**
While creating the account and login pages for a craft project I used `{% requireLogin %}` on account pages but there is nothing similar to redirect an authenticated user away from a login page. 

When adding `{% requireGuest %}` to login/register templates, authenticated users will get redirected away.

### CODE REVIEW

File: src/web/twig/nodes/RequireGuestNode.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  /**
+   * Compiles a RequireLoginNode into PHP.
+   *
+   * @param Compiler $compiler
+   */
+  public function compile(Compiler $compiler)
+  {
+    $compiler
+        ->addDebugInfo($this)
+        ->write(Craft::class . "::\$app->controller->requireGuest();\n");
```
Comment (brandonkelly) [inline src/web/twig/nodes/RequireGuestNode.php]:
```suggestion
        ->write(Craft::class . "::\$app->getUser()->guestRequired();\n")
        ->write(Craft::class . "::\$app->end();\n");
```

File: src/web/Controller.php [hunk 2]
```diff
@@ -261,6 +261,21 @@ public function requireLogin()
...
+     * Redirects the user to the account template if they are logged in.
+     */
+    public function requireGuest()
+    {
+        $userSession = Craft::$app->getUser();
+
+        if (!$userSession->getIsGuest()) {
+            return Craft::$app
+                ->getResponse()
+                ->redirect(Craft::$app->user->returnUrl);
```
Comment (brandonkelly) [inline src/web/Controller.php]: This logic should probably live on `craft\web\User` in a `guestRequired()` method, similar to `lognRequired()`.

File: src/web/twig/nodes/RequireGuestNode.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+use Craft;
+use Twig\Compiler;
+use Twig\Node\Node;
+
+/**
+ * Class RequireLoginNode
+ *
+ * @author Pixel & Tonic, Inc. <support@pixelandtonic.com>
+ * @since 3.0
```
Comment (brandonkelly) [inline src/web/twig/nodes/RequireGuestNode.php]:
```suggestion
 * @since 3.4.0
```

File: src/web/twig/nodes/RequireGuestNode.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+namespace craft\web\twig\nodes;
+
+use Craft;
+use Twig\Compiler;
+use Twig\Node\Node;
+
+/**
+ * Class RequireLoginNode
```
Comment (brandonkelly) [inline src/web/twig/nodes/RequireGuestNode.php]:
```suggestion
 * Class RequireGuestNode
```

File: src/web/twig/nodes/RequireGuestNode.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ * @author Pixel & Tonic, Inc. <support@pixelandtonic.com>
+ * @since 3.0
+ */
+class RequireGuestNode extends Node
+{
+  // Public Methods
+  // =========================================================================
+
+  /**
+   * Compiles a RequireLoginNode into PHP.
```
Comment (brandonkelly) [inline src/web/twig/nodes/RequireGuestNode.php]:
```suggestion
   * Compiles a RequireGuestNode into PHP.
```

File: src/web/twig/tokenparsers/RequireGuestTokenParser.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace craft\web\twig\tokenparsers;
+
+use craft\web\twig\nodes\RequireGuestNode;
+use Twig\Parser;
+use Twig\Token;
+use Twig\TokenParser\AbstractTokenParser;
+
+/**
+ * Class RequireLoginTokenParser
```
Comment (brandonkelly) [inline src/web/twig/tokenparsers/RequireGuestTokenParser.php]:
```suggestion
 * Class RequireGuestTokenParser
```

File: src/web/twig/tokenparsers/RequireGuestTokenParser.php [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * @author Pixel & Tonic, Inc. <support@pixelandtonic.com>
+ * @since 3.0
```
Comment (brandonkelly) [inline src/web/twig/tokenparsers/RequireGuestTokenParser.php]:
```suggestion
 * @since 3.4.0
```

File: src/web/Controller.php [hunk 8]
```diff
@@ -261,6 +261,21 @@ public function requireLogin()
         }
     }
 
+    /**
+     * Redirects the user to the account template if they are logged in.
```
Comment (brandonkelly) [inline src/web/Controller.php]:
```suggestion
     * Redirects the user to the account template if they are logged in.
     *
     * @since 3.4.0
```

**Comment by wrux** [comment]
> Thanks for the PR! Made a few comments & suggestions.

I made some changes and used `yii\web\User::loginRequired()` and `craft\web\Controller::requireLogin()` as a reference.

**Comment by brandonkelly** [comment]
Tweaked guestRequired() a bit but otherwise looks good. All merged in for v3.4, thanks!

**Comment by wrux** [comment]
Great, thanks. Also I know it’s super minor, but the commit shows a different user to my github username. Is there any way to change that?

**Comment by brandonkelly** [comment]
What do you mean? Both of your commits show they were committed by @wrux (e3566ff922a6d24b51e069bc79bd5770af5005ae & 13af178f3e8a19ea127b2572de20c0ef2fa42e56).

**Pull request merged**